apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component, PropTypes } from 'react';
+import _ from 'lodash';
+import { assignmentShape } from './shapes';
+
+export default class AssignmentSelector extends Component {
+  static propTypes = {
+    courseId: PropTypes.number,
+    scriptId: PropTypes.number,
+    validCourses: PropTypes.arrayOf(assignmentShape).isRequired,
+    validScripts: PropTypes.arrayOf(assignmentShape).isRequired,
+  };
+
+  getSelectedAssignment() {
+    const assignment = this.assignments[this.root.value];
+    return {
+      course_id: assignment.course_id,
+      script_id: assignment.script_id
+    };
+  }
+
+  render() {
+    const { courseId, scriptId, validCourses, validScripts } = this.props;
+
+    // TODO(bjvanminnen): It's not clear that all of this data manipulation belongs
+    // here. If we move our data to redux, it likely belongs there. Otherwise, it
+    // may still make more sense for this to all happen before instantiating our
+    // React tree.
+
+    // Differentiate courses and ids by giving them course_ids and script_ids fields.
+    const courses = validCourses.map(course => ({
+      ...course,
+      course_id: course.id
+    }));
+    const scripts = validScripts.map(script => ({
+      ...script,
+      script_id: script.id
+    }));
+
+    // concat courses and scripts, givin them an index to that we can easily
+    // get back to assignment.
+    const assignments = courses.concat(scripts).map((assignment, index) => ({
+      ...assignment,
+      index
+    }));
+
+    // store as instance variable so that getSelectedAssignment doesnt need to
+    // regenerate
+    this.assignments = assignments;
+
+    let currentAssignment = '';
+    if (courseId) {
+      const selectedCourse = validCourses.findIndex(c => c.id === courseId);
+      if (selectedCourse !== -1) {
+        currentAssignment = selectedCourse;
+      }
+    } else if (scriptId) {
+      const selectedScript = validScripts.findIndex(s => s.id === scriptId);
+      currentAssignment = validCourses.length + selectedScript;
+    }
+
+    const grouped = _.groupBy(
+      _.orderBy(assignments, ['category_priority', 'category', 'position', 'name']),
+      'category'
+    );
```

Comment:
There's a lot of stuff going on here that I don't love. Some of this comes about because I don't want to change what the server is returning for valid_courses/valid_scripts, because that will also affect our angular app. Most of it I think will end up feeling cleaner once I add redux and am able to put a bunch of this logic into our redux module.